learning_observer/learning_observer/canvas.py

```diff
@@ -0,0 +1,308 @@
+import os
+import json
+import string
+import recordclass
+import configparser
+import aiohttp
+import aiohttp.web
+
+import learning_observer.settings as settings
+import learning_observer.log_event
+import learning_observer.util
+import learning_observer.auth
+import learning_observer.runtime
+
+cache = None
+
+class Endpoint(recordclass.make_dataclass("Endpoint", ["name", "remote_url", "doc", "cleaners"], defaults=["", None])):
```

Comment:
What of this can be abstracted out? We use a similar approach for both Google and Schoology. We don't want to repeat code 3 times.